lte/gateway/c/core/oai/tasks/mme_app/mme_config.cpp

```diff
@@ -1513,12 +1516,16 @@ int mme_config_parse_string(const char* config_string,
               config_pP->nas_config.prefered_integrity_algorithm[i] =
                   EIA2_128_ALG_ID;
             else
-              config_pP->nas_config.prefered_integrity_algorithm[i] =
-                  EIA0_ALG_ID;
+              Fatal(
+                  "Unrecognized algorithm in "
+                  "ORDERED_SUPPORTED_INTEGRITY_ALGORITHM_LIST\n");
           }
 
+          char default_integrity_algorithm =
+              config_pP->nas_config.prefered_integrity_algorithm[num - 1];
           for (i = num; i < 8; i++) {
-            config_pP->nas_config.prefered_integrity_algorithm[i] = EIA0_ALG_ID;
+            config_pP->nas_config.prefered_integrity_algorithm[i] =
+                default_integrity_algorithm;
```

Comment:
Hi @lucasgonze, Instead of filling the last configured value from the list. Can we set to invalid value say "0b111" 
I think the similar are required for encryption/ciphering algorithm, at https://github.com/magma/magma/pull/14468/files#diff-108a949404fe1a919b6e0e91872fcdf52cbfb1c47fba2066c1a5267734ca7e61R1558

Reply:
I was under impression that '0b111' would be invalid value, but referring to spec indicated that this value maps to NAS_SECURITY_ALGORITHMS_EEA7. Thanks for instruct me to rethink on this.
Can we configure last configured value from the list, other than EIA0_ALG_ID. Because we would to avoid configuring "null encryption"

Reply:
Good teamwork.

I am not sure I am understanding your point correctly. The code as it stands is using the last configured value from the list, rather than defaulting to EIA0_ALG_ID.

Reply:
If integrity is configured with, ORDERED_SUPPORTED_INTEGRITY_ALGORITHM_LIST = [ "EIA2" , "EIA1", "EIA0" ]; then remaining indices of "config_pP->nas_config.prefered_integrity_algorithm" would be filled with "EIA0_ALG_ID". 
We will set the remaining indices with last configured but other than "EIA0_ALG_ID".
Considering above configuration, the last configured value is "EIA0",instead of this we can set  "EIA1".
Correct me if I am wrong.

Reply:
So the code would be something like

`        ORDERED_SUPPORTED_INTEGRITY_ALGORITHM_LIST = [ "EIA2" , "EIA0" , "EIA1" ];`

instead of the current:

`        ORDERED_SUPPORTED_INTEGRITY_ALGORITHM_LIST = [ "EIA2" , "EIA1" , "EIA0" ];`

Reply:
I'm checking out the impact in the 3gpp specs. Here's the document:
https://www.etsi.org/deliver/etsi_ts/133400_133499/133401/16.03.00_60/ts_133401v160300p.pdf

The section is 7.2.4.3 "Procedures for NAS algorithm selection".

Reply:
Okay got it. As per given reference above.  The MME selects the
NAS algorithms which have the highest priority according to the ordered lists.
So in our code, the remaining indices needs to be filled with last configured value. I agree with whatever changes you have done is correct.
Thanks for providing the reference.